src/components/ArrowKeyFocusManager.js

```diff
@@ -0,0 +1,139 @@
+import _ from 'underscore';
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+import * as PropTypeUtils from '../libs/PropTypeUtils';
+import CONST from '../CONST';
+import KeyboardShortcut from '../libs/KeyboardShortcut';
+
+const propTypes = {
+    /** Children to render. */
+    children: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.node,
+    ]).isRequired,
+
+    /** The initial focused index. */
+    initialFocusedIndex: PropTypeUtils.integerPropType,
+
+    /** The number of items in the list that this component is managing focus for. */
+    listLength: PropTypeUtils.wholeNumberPropType,
+
+    /** A callback executed when the focused input changes. */
+    onFocusedIndexChanged: PropTypes.func,
+
+    /** A callback executed when the enter key is pressed. */
+    onEnterKeyPressed: PropTypes.func,
+
+    /** Should the enter key event bubble? */
+    shouldEnterKeyEventBubble: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
+};
+
+const defaultProps = {
+    initialFocusedIndex: 0,
+    listLength: 0,
+    onFocusedIndexChanged: () => {},
+    onEnterKeyPressed: undefined,
+    shouldEnterKeyEventBubble: false,
+};
+
+class ArrowKeyFocusManager extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            focusedIndex: props.initialFocusedIndex,
+        };
+        this.shouldEnterKeyEventBubble = this.shouldEnterKeyEventBubble.bind(this);
+    }
+
+    componentDidMount() {
+        const enterConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
+        const arrowUpConfig = CONST.KEYBOARD_SHORTCUTS.ARROW_UP;
+        const arrowDownConfig = CONST.KEYBOARD_SHORTCUTS.ARROW_DOWN;
+
+        if (this.props.onEnterKeyPressed) {
```

Comment:
Oh I see I guess it's defaulted to `undefined` because of this line?

Reply:
yep!